Tests/AWSLambdaRuntimeCoreTests/LambdaMockClient.swift

```diff
@@ -0,0 +1,271 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import AWSLambdaRuntimeCore
+import Foundation
+import Logging
+import NIOCore
+
+struct LambdaMockWriter: LambdaResponseStreamWriter {
+    var underlying: LambdaMockClient
+
+    init(underlying: LambdaMockClient) {
+        self.underlying = underlying
+    }
+
+    mutating func write(_ buffer: ByteBuffer) async throws {
+        try await self.underlying.write(buffer)
+    }
+
+    func finish() async throws {
+        try await self.underlying.finish()
+    }
+
+    func writeAndFinish(_ buffer: ByteBuffer) async throws {
+        try await self.underlying.write(buffer)
+        try await self.underlying.finish()
+    }
+
+    func reportError(_ error: any Error) async throws {
+        try await self.underlying.write(ByteBuffer(string: "\(error)"))
+        try await self.underlying.finish()
```

Comment:
I think this is wrong. It should trigger a reportError function. And that reportError function should fail the Continuation in `handlerIsProcessing`